x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/utils/indices.ts

```diff
@@ -25,16 +25,28 @@ import {
   IngestionStatus,
 } from '../types';
 
+const CRAWLER_CONNECTOR_TYPE = 'elastic-crawler';
+
 export function isConnectorIndex(
   index: ElasticsearchIndexWithIngestion | undefined
 ): index is ConnectorIndex {
-  return !!(index as ConnectorIndex)?.connector;
+  const indexAsConnector = index as ConnectorIndex;
+  return (
+    !!indexAsConnector?.connector &&
+    indexAsConnector.connector.service_type !== CRAWLER_CONNECTOR_TYPE
+  );
 }
 
 export function isCrawlerIndex(
   index: ElasticsearchIndexWithIngestion | undefined
 ): index is CrawlerIndex {
-  return !!(index as CrawlerIndex)?.crawler;
+  const indexAsConnector = index as ConnectorIndex;
+  const indexAsCrawler = index as CrawlerIndex;
+  return (
+    !!indexAsCrawler?.crawler ||
+    (!!indexAsConnector?.connector &&
```

Comment:
This has the potential to break our types. This guard function will now return that the index has a `crawler` key even if it does not. That is because the [interface CrawlerIndex](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/common/types/indices.ts#L39-L41) says that `crawler` key will be set.

You likely want a different utility function that is similar to this but doesn't have the `index is CrawlerIndex` guard.

Reply:
Or we could update the `CrawlerIndex` type to be a union that can have `crawler` or `connector`. That would be something like:

```typescript
export type CrawlerIndex =  ElasticsearchIndex & {
  crawler: Crawler;
} | ElasticsearchIndex & {
  connector: Connector & {
    service_type: 'elastic-crawler';
  }
}
```

But this will likely require a lot of changes in code that uses `CrawlerIndex`.

Reply:
Thanks for the review @TattdCodeMonkey !

Sorry my typescript experience is zero + this PR, so I didn't realise the guard clause was so important. Are you saying that I can fix the type-breaking danger by updating the guard clause?

On your second point, I'm hoping to keep the changes localised to this file so I'd rather go with the first option (updating guard clause). WDYT?

Reply:
Strongly agree with @TattdCodeMonkey that we don't want to typecast to Crawler--that'll lead to more issues down the road. 

Ideally the move to registering as a connector replaces the current use of Crawler, which means that we can just replace the old type with a new type. I am not sure how you implemented this on the backend, but I assume that the index object we get back from the backend should now contain *both* a crawler object *and* a connector object? If not, we should fix the server so it does--that would solve all of these problems.

Reply:
@navarone-feekery The important part is ensuring the types are accurate. So if the backend returns an object with both `crawler` and `connector` keys for an `elastic-crawler` then we could introduce a new interface:

```typescript
export interface ElasticCrawlerIndex extends ElasticsearchIndex {
  crawler: Crawler;
  connector: Connector;
}
```

Then introduce a new `isElasticCrawlerIndex` function. Or if we want to retain usage of only the `isCrawlerIndex` then we would want to update the `CrawlerIndex`

```typescript
export interface CrawlerIndex extends ElasticsearchIndex {
  crawler: Crawler;
  connector?: Connector;
}
```

But if we want to treat an index that has `connector.service_type: 'elastic_crawler'` as a crawler without a `crawler` key, then that is not typesafe. We would instead need to introduce a new type and handle that type anywhere we check for a `CrawlerIndex`. And that change would be more than just this file.

Reply:
In terms of functionality this new crawler *should* have a crawler object returned. Otherwise eh...none of the crawler UIs will work anyway, heh. It may be that this requires some Kibana server changes though, I haven't checked how we're retrieving the crawler data right now.

Reply:
A quick peek at it suggests the crawler property should be present (you should be able to confirm with a peek at the network tab). In that case @TattdCodeMonkey's second suggestion (adding a connector property to the CrawlerIndex type) is probably best.